docs/extensibility/interaction-service.md

```diff
@@ -0,0 +1,359 @@
+>
+> For example:
+>
+> ```csharp
+> var builder = DistributedApplication.CreateBuilder(args);
+> 
+> builder.AddFakeResource("fake-resource")
+>        .WithCommand("msg-dialog", "Example Message Dialog", async context =>
+> {
+>     var interactionService = context.GetRequiredService<IInteractionService>();
+>     
+>     // Use interaction service...
+>
+>     return CommandResults.Success();
+> });
+> ```
+>
+> For CLI specific contexts, the interaction service is retrieved from either the `PublishingContext` or `DeploymentContext` depending on the operation being performed.
+
+## Display messages
+
+There are several ways to display messages to the user:
+
+- **Dialog messages**: Show important information in a dialog box.
+- **Notification messages**: Display less critical information in a notification.
+
+> [!NOTE]
+> Message display methods (`PromptMessageBoxAsync` and `PromptNotificationAsync`) are only available in dashboard contexts. These methods throw an exception if called during `aspire publish` or `aspire deploy` operations.
+
+### Display a dialog message box
+
+Dialog messages provide important information that requires user attention.
+
+<!-- <xref:Aspire.Hosting.IInteractionService.PromptMessageBoxAsync%2A> -->
+
+The `IInteractionService.PromptMessageBoxAsync` method displays a message with customizable response options.
+
+:::code source="snippets/InteractionService/AppHost.MessageBoxExample.cs" id="example":::
+
+**Dashboard view:**
+
+:::image type="content" source="media/interaction-service-message-dialog.png" lightbox="media/interaction-service-message-dialog.png" alt-text="Aspire dashboard interface showing a message dialog with a title, message, and buttons.":::
+
+**CLI view:**
+
+The `PromptMessageBoxAsync` method only works in the dashboard context. If you call it during `aspire publish` or `aspire deploy`, the method throws an exception and doesn't display a message in the CLI.
+
+### Display a notification message
+
+Notification messages provide nonmodal notifications.
+
+> [!TIP]
+> In the dashboard, notification messages appear stacked at the top, so you can show several messages at once. You can display notifications one after another by awaiting each dismissal before showing the next. Or, you can display multiple notifications at the same time without waiting for each to be dismissed.
+
+<!-- <xref:Aspire.Hosting.IInteractionService.PromptNotificationAsync%2A> -->
+
+The `IInteractionService.PromptNotificationAsync` method displays informational messages with optional action links in the dashboard context. You don't have to await the result of a notification message if you don't need to. This is especially useful for notifications, since you might want to display them and continue without waiting for user interaction.
```

Comment:
```suggestion
The `IInteractionService.PromptNotificationAsync` method displays informational messages with optional action links in the dashboard context. You don't have to await the result of a notification message if you don't need to. This is especially useful for notifications, since you might want to display a notification and continue without waiting for user to dismiss it.
```